### PR TITLE
test: extend timeout for 20-main-project.md

### PR DESCRIPTION
**Pull request by jpivarski**
The scheduled tests failed for the past 12 days in a row, but for a variety of reasons:

```
2025-01-08	03-basic-fitting.md     assert np.sum((y_from_ansatz - y)**2) < 100
2025-01-09	24-convolutional.md     assert confusion_matrix[0:2, 2:].sum() / confusion_matrix[0:2].sum() < 0.5
2025-01-10      24-convolutional.md     assert confusion_matrix[0:2, 2:].sum() / confusion_matrix[0:2].sum() < 0.5
2025-01-11      20-main-project.md      A cell timed out while it was being executed, after 120 seconds.
2025-01-12      20-main-project.md      A cell timed out while it was being executed, after 120 seconds.
2025-01-13      20-main-project.md      A cell timed out while it was being executed, after 120 seconds.
2025-01-14      20-main-project.md      A cell timed out while it was being executed, after 120 seconds.
2025-01-15      20-main-project.md      A cell timed out while it was being executed, after 120 seconds.
2025-01-16      03-basic-fitting.md     assert np.sum((y_from_ansatz - y)**2) < 100
2025-01-17      20-main-project.md      A cell timed out while it was being executed, after 120 seconds.
2025-01-18      20-main-project.md      A cell timed out while it was being executed, after 120 seconds.
2025-01-19      04-universal-approximators.md  assert np.sum((model_y - curve_y)**2) < 10
```

The most common of these is that the cell execution timeout on 20-main-project.md is too short. It's reasonable to make that longer, so this PR extends it to 10 minutes (overkill). The assertions, however, are important for ensuring that the web pages show correct results from a fitter whose initial state is non-deterministic—Minuit in all of the above cases. The only way the assertions could be improved would be by setting a PRNG seed in iminuit or set the initial fit parameters closer to the minimum (which undermines the lesson). I don't see a way to set the seed in iminuit, but if anyone has any ideas, that would be great.

Meanwhile, I'm only addressing the timeouts in this PR. We should see the tests pass more often, but not every day.

### CODE REVIEW

**Comment by jpivarski** [comment]
Ah! openml.org is inaccessible: https://github.com/orgs/openml/discussions/20

All of the failures in the past week or so can be blamed on that, though on some days, different (expected intermittent) issues occurred before the openml.org one could happen. It looks like they'll be restarting the server tomorrow, so it doesn't make sense for me to develop a work-around. I'll just have to remember to trigger the tests again tomorrow.

**Comment by jpivarski** [comment]
I checked again today and although the situation is looking better, the data-fetch still doesn't work.

**Comment by jpivarski** [comment]
The site is sufficiently online for me to see the [description page](https://openml.org/search?type=data&sort=version&status=any&order=asc&exact_name=hls4ml_lhc_jets_hlf&id=42468), but if you click the [download button](https://openml.org/datasets/0004/4246/dataset_42468.arff), you get

```xml

<Error>
    <Code>NoSuchKey</Code>
    <Message>The specified key does not exist.</Message>
    <Key>0004/4246/dataset_42468.arff</Key>
    <BucketName>datasets</BucketName>
    <Resource>/datasets/0004/4246/dataset_42468.arff</Resource>
    <RequestId>181D5B2FF5DA20E1</RequestId>
    <HostId>dd9025bab4ad464b049177c95eb6ebf374d3b3fd1af9251148b658df7ac2e3e8</HostId>
</Error>
```

The script

```python
import sklearn.datasets
d = sklearn.datasets.fetch_openml("hls4ml_lhc_jets_hlf")
d["data"], d["target"]
```

tries to fetch it through URL `https://api.openml.org/data/v1/download/21854360`, which raises HTTP error 404 (not found).

So we're still waiting.

**Comment by jpivarski** [comment]
Since we're waiting on an upstream server, I'm going to merge this PR so that everything in it gets tested daily, rather than when I pull the trigger. It's not making anything worse, not introducing wrong code into `main` (unless it was wrong to update the GitHub Actions, but that's unlikely because of [this example](https://github.com/actions/upload-pages-artifact?tab=readme-ov-file#usage).
